quantum/quantum.c

```diff
@@ -837,11 +837,12 @@ bool process_record_quantum(keyrecord_t *record) {
 }
 
 __attribute__ ((weak))
-const bool ascii_to_shift_lut[0x80] PROGMEM = {
+const bool ascii_to_shift_lut[128] PROGMEM = {
```

Comment:
Why the change from 0x80 to 128?

Reply:
I just prefer decimal in array lengths.

Reply:
If anything this could be a define since it's a magic number

Reply:
Some layouts might want to map extended ASCII characters like £ (UK layout) so having a variable LUT length could be nice. Does the weak attribute allow for different array lengths here?

Reply:
GCC doesn't complain when I omit the length from the LUT declarations in quantum.h, though I'm wary of what would happen if sendstring tried to look up an extended ASCII code in the default shift/altgr tables (some layouts don't need them).

Reply:
The better question would be if it worked.  

Also, if the shift/alt tables handle that, then do we need to extend it?

Reply:
It seems that `send_char` doesn't do any sanity checks on the LUT indexes:

https://github.com/qmk/qmk_firmware/blob/7dcf9237d5cdf83e6f357755c19c14e92e6593c0/quantum/quantum.c#L837-L839

So I'm guessing you can already persuade `ascii_code` to be 129, and... I don't know what happens then. It reads some other byte next to the LUT, maybe.

Reply:
Probably not.  If it's outside of the array, it should error out.  But you would want to make sure the array is set correctly, in that case. 

At least, IIRC.